File: Sesion de clase 15 - ProgramacionOrientadaObjetos/Cilindro.cpp

#include "Circulo.h"
#include "Cilindro.h"

Cilindro::Cilindro(double r, double h) : Circulo(r), h(h) {
//    this->h = h;
}

double Cilindro::calcularArea() {
    return Circulo::calcularArea() * this->h;
}

File: ejercicios-java/clase-04/Cilindro.java
// interface Runnable {
// 	void run();
// }

// class Semaforo implements Runnable {
// 	@Override
// 	public void run() {
// 		//TODO: se coloca el código del hilo
// 	}
// }

// class Cilindro extends Circulo implements Objecto {
class Cilindro extends Circulo {
	private double altura;

	public Cilindro(String identificacion, double radio, double altura) {
		super(identificacion, radio);
		this.altura = altura;
	}

	public double getAltura() {
		return altura;
	}

	public void setAltura(double altura) {
		this.altura = altura;
	}

	@Override
	public double area() {
		return 2 * super.area() + perimetro() * altura;
	}

	public double areaDeLaBase() {
		return super.area();
	}

	public double volumen() {
		return super.area() * altura;
	}
}